gravitee-am-gateway/gravitee-am-gateway-standalone/gravitee-am-gateway-standalone-container/src/main/java/io/gravitee/am/gateway/vertx/VertxServerConfiguration.java

```diff
@@ -41,31 +42,42 @@ public class VertxServerConfiguration {
     @Lazy
     @Bean
     public VertxHttpServerOptions httpServerConfiguration(
-            Environment environment,
-            KeyStoreLoaderManager keyStoreLoaderManager
+            Environment environment
     ) {
         return VertxHttpServerOptions.builder()
                 .prefix(HTTP_PREFIX)
                 .environment(environment)
                 .port(environment.getProperty(HTTP_PREFIX + ".port", Integer.class, 8092))
-                .keyStoreLoaderManager(keyStoreLoaderManager)
                 .id(HTTP_PREFIX)
                 .build();
     }
 
-    @Bean("gatewayHttpServer")
-    @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
-    public VertxHttpServer vertxHttpServerFactory(Vertx vertx, VertxHttpServerOptions options) {
-        return new VertxHttpServerFactory(vertx).create(options);
+
+    @Bean
+    public ServerManager serverManager(
+            VertxServerFactory<VertxServer<?, VertxServerOptions>, VertxServerOptions> serverFactory,
+            Environment environment
+    ) {
+        final DefaultServerManager serverManager = new DefaultServerManager();
+        final VertxHttpServerOptions options = VertxHttpServerOptions.builder()
+                .prefix(HTTP_PREFIX)
+                .environment(environment)
+                .port(environment.getProperty(HTTP_PREFIX + ".port", Integer.class, 8092))
+                .id(HTTP_PREFIX)
+                .build();
```

Comment:
**suggestion**: If possible maybe we can pass the VertxHttpServerOptions provided by the @Bean method line 45 as parameter to mutualize this instantiation ? :thinking: